orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.groovy

```diff
@@ -490,8 +494,12 @@ class JedisExecutionRepository implements ExecutionRepository {
       keepWaitingPipelines: String.valueOf(execution.keepWaitingPipelines),
       executionEngine     : execution.executionEngine?.name() ?: DEFAULT_EXECUTION_ENGINE.name()
     ]
-    // TODO: store separately? Seems crazy to be using a hash rather than a set
     map.stageIndex = execution.stages.id.join(",")
+    // TODO: remove this and only use the list
+    if (!execution.stages.empty) {
+      jedis.del("$key:stageIndex")
+      jedis.rpush("$key:stageIndex", *execution.stages.id)
+    }
```

Comment:
dual-write when storing the whole execution. Note this is still not concurrency safe but the intention is to not be calling this method all the time anyway